pkg/artifactory/resource_artifactory_remote_repository.go

```diff
@@ -517,7 +522,12 @@ func resourceRemoteRepositoryDelete(d *schema.ResourceData, m interface{}) error
 		return err
 	}
 	resp, err := c.V1.Repositories.DeleteRemote(context.Background(), *repo.Key)
-	if resp.StatusCode == http.StatusNotFound {
+
+	if resp != nil {
```

Comment:
The equality check should be `resp == nil` but check `err` first. If you check error first you probably never need to check if `resp` is `nil`.